pkg/tenant/db/organizations.go

```diff
@@ -0,0 +1,72 @@
+package db
+
+import (
+	"context"
+
+	"github.com/oklog/ulid/v2"
+	"github.com/rotationalio/ensign/pkg/utils/ulids"
+)
+
+const OrganizationNamespace = "organizations"
+
+// Use the resourceID to retrieve the orgID from the database.
+func GetOrgIndex(ctx context.Context, resourceID ulid.ULID) (orgID ulid.ULID, err error) {
+	if ulids.IsZero(resourceID) {
+		return ulid.ULID{}, ErrMissingID
+	}
+
+	var data []byte
+	if data, err = getRequest(ctx, OrganizationNamespace, resourceID[:]); err != nil {
+		return ulid.ULID{}, err
+	}
+
+	if err = orgID.UnmarshalBinary(data); err != nil {
+		return ulid.ULID{}, err
+	}
+	return orgID, nil
+}
+
+// Store the resourceID as a key and the orgID as a value in the database.
+func PutOrgIndex(ctx context.Context, resourceID, orgID ulid.ULID) error {
+	if ulids.IsZero(resourceID) {
+		return ErrMissingID
+	}
+
+	if ulids.IsZero(orgID) {
+		return ErrMissingOrgID
+	}
+
+	if err := putRequest(ctx, OrganizationNamespace, resourceID[:], orgID[:]); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// VerifyOrg will check that resources are allocated to the correct organization.
+// The method will take in an orgID from the claims and will return true if the orgID
+// from the database is the same and an error if it is not.
+func VerifyOrg(ctx context.Context, claimsOrgID, resourceID ulid.ULID) (bool, error) {
+	if ulids.IsZero(claimsOrgID) {
+		return false, ErrMissingOrgID
+	}
+
+	if ulids.IsZero(resourceID) {
+		return false, ErrMissingID
+	}
+
+	orgID, err := GetOrgIndex(ctx, resourceID)
+	if err != nil {
+		return false, err
+	}
+
+	err = PutOrgIndex(ctx, resourceID, orgID)
+	if err != nil {
+		return false, err
+	}
+	if orgID.Compare(claimsOrgID) == 0 {
+		return true, nil
+	} else {
+		return false, ErrOrgNotVerified
```

Comment:
@pdeziel do you want an error here in addition to the false?

This is kind of a chicken and an egg situation; the `err` implies that verification failed; but in this case verification "succeeded" in the sense that we proved that the org is not verified. So really it depends on how you want to perform this check. E.g. if you do

```go
if verified, err := VerifyOrg(ctx, orgID, resourceID); err != nil || !verified {
	return ErrNotAuthorized
}
```

Then it implies that `false` and an error will always be returned (which makes the bool a bit redundant). The other option is:

```go
if verified, err = VerifyOrg(ctx, orgID, resourceID); err != nil {
    // There was a database error
   sentry.Error(ctx).Err(err).Msg("could not check verification")
   return ErrInternal
}

if !verified {
   // Not verified
   return ErrNotAuthorized
}
```

Then the bool and the error make sense together; but we shouldn't return an error in the `false` case.

Reply:
I think the way I've been handling these situations is to just return the error since it reduces the redundancy and simplifies the interface. However, we can add an error code check if we want to do conditional error logging.

```golang
if err = VerifyOrg(ctx, orgID, resourceID); err != nil {
    if !errors.Is(ErrNotFound) {
        sentry.Error(ctx).Err(err).Msg("could not check verification")
        return err
    }
    return err
}
```

Reply:
Ok perfect; I'd just change it to a `sentry.Warn` instead of a `sentry.Error` but that works for me.